### PR TITLE
Allow construction of app/device objects from json

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds non-public methods for constructing `BugsnagAppWithState` and `BugsnagDeviceWithState` from JSON. These will be used by React Native when reading JS payloads.

### CODE REVIEW

File: Source/BugsnagDeviceWithState.m [hunk 1]
```diff
@@ -75,6 +75,33 @@ - (instancetype)init {
...
+    device.freeMemory = json[@"freeMemory"];
+    device.freeDisk = json[@"freeDisk"];
+    device.locale = json[@"locale"];
+    device.manufacturer = json[@"manufacturer"];
+    device.model = json[@"model"];
+    device.modelNumber = json[@"modelNumber"];
+    device.orientation = json[@"orientation"];
+    device.osName = json[@"osName"];
+    device.osVersion = json[@"osVersion"];
+    device.runtimeVersions = json [@"runtimeVersions"];
```
Comment (tomlongridge) [inline Source/BugsnagDeviceWithState.m:90]:
```suggestion
    device.runtimeVersions = json[@"runtimeVersions"];
```

File: Source/BugsnagDeviceWithState.m [hunk 2]
```diff
@@ -75,6 +75,33 @@ - (instancetype)init {
...
+    device.totalMemory = json[@"totalMemory"];
+
+    id jailbroken = json[@"jailbroken"];
+    if (jailbroken) {
+        device.jailbroken = [(NSNumber *) jailbroken boolValue];
```
Comment (tomlongridge) [inline Source/BugsnagDeviceWithState.m:95]: Jailbroken is not listed as nullable - do we need a default?

File: Source/BugsnagAppWithState.m [hunk 3]
```diff
@@ -22,6 +22,34 @@ - (NSDictionary *)toDict;
...
+
+    id durationInForeground = json[@"durationInForeground"];
+    if (durationInForeground && [durationInForeground isKindOfClass:[NSNumber class]]) {
+        app.durationInForeground = [(NSNumber *) durationInForeground unsignedIntValue];
+    }
+
+    id inForeground = json[@"inForeground"];
+    if (inForeground) {
+        app.inForeground = [(NSNumber *) inForeground boolValue];
+    }
```
Comment (tomlongridge) [inline Source/BugsnagAppWithState.m:41]: These properties aren't nullable - do we need a default?

File: Source/BugsnagDeviceWithState.m (hunk 2 above)
Comment (fractalwrench) [reply Source/BugsnagDeviceWithState.m:95]: The default will be false for both the `jailbroken` and `inForeground` properties as that's the default value for the `BOOL` type in Objective-C.

**Pull request merged**